Related items start

markmiw opened the pull request.

markmiw committed:
Completed basic requirements

markmiw committed:
Completed Outfit Section

markmiw committed:
stopped tracking .env

markmiw committed:
Updated Styles

markmiw committed:
Updated gitignore

markmiw committed:
removed dist

markmiw committed:
merged
Merge branch 'main' into related_items_start

JeromeMTR commented on client/src/App.jsx:
```diff
@@ -1,14 +1,109 @@
+
+  return (
+    <div className="app">
+      {Object.keys(product).length !== 0 ? [
+        <Related product={product.data} updateCurrentProduct={updateCurrentProduct}/>,
+        <Outfit productStyle={productStyle.data} outfits={outfits} removeOutfit={removeOutfit} addToOutfit={addToOutfit} carouselPos={carouselPos}/>,
+      ] : null}
```
@markmiw the Related and Outfit component need a surrounding div or unique key prop

JeromeMTR commented on client/src/Related/RelatedModal.jsx:
```diff
@@ -0,0 +1,73 @@
+        }
+        // if they have the same characteristic and value
+        let compCharacteristic = characteristics[characteristic][0];
+        if (compCharacteristic === value) {
+          characteristics[characteristic][0] = '&#10003';
+          characteristics[characteristic][1] = characteristic;
+          characteristics[characteristic][2] = '&#10003';
```
@markmiw on line 41 and 43 its printing out a string, also it is saying they match but is not describing what the value of matched characteristics
<img width="321" alt="Screen Shot 2022-07-13 at 3 10 54 PM" src="https://user-images.githubusercontent.com/101430253/178845536-d8b450f2-f313-4b6c-80ad-63e410597a2c.png">

JeromeMTR commented on client/src/Related/RelatedCard.jsx:
```diff
@@ -1,16 +1,69 @@
+          showModal(e, id);
+        }}
+        />
+        <div onClick={(e) => {
+          updateCurrentProduct(e, id);
+        }}className="description">
+          <p>{category}</p>
+          <b>{name}</b>
+          <p>${Math.round(originalPrice)}</p>
+          {rating ? <StarRating rating={rating} /> : null}
+        </div>
```
@markmiw There is a bug when there are more than 4 related products. When clicking the 5th or greater products in carousel it creates a bug if that product has more than 4 related items

JeromeMTR commented on client/src/Related/Outfit.jsx:
```diff
@@ -0,0 +1,69 @@
+    element.classList.remove(curPosition);
+    element.classList.add(newPosition);
+  }
+  return (
+    <div className="related">
+      <h5 className="title">YOUR OUTFIT</h5>
+      <div className="carousel">
+        <button onClick={(e) => addToOutfit(e, Object.keys(productStyle.data)[0])} className="add-outfit">{'+'}</button>
+        <button onClick={(e) => moveOutfitLeft(e)} className="carousel-prev">{'<'}</button>
+        <div className="carousel-inner">
+          {Object.keys(outfits).map((id, index) => <OutfitCard
+            key={id}
+            addToOutfit={addToOutfit}
+            removeOutfit={removeOutfit}
+            id={id}
+            outfit={outfits[id]}
+            position={carouselPos[id]} />)}
+        </div>
+        <button onClick={(e) => moveOutfitRight(e)} className="carousel-next">{'>'}</button>
```
There seems to be another bug when there are more than 4 outfits and you start deleting and clicking left and right buttons to shift outfits.  Also at times it meges cards together

markmiw committed:
Updated bug. Still fixing carousel bug

markmiw committed:
Tries
Merge branch 'main' of https://github.com/Team-Pacman/Rhydon into related_items_start
